api/server.go

```diff
@@ -61,63 +61,6 @@ func (s *Server) Ping(_ context.Context, _ *access.PingRequest) (*access.PingRes
 	return &access.PingResponse{}, nil
 }
 
-// GetLatestBlockHeader implements the GetLatestBlockHeader endpoint from the Flow Access API.
-// See https://docs.onflow.org/access-api/#getlatestblockheader
-func (s *Server) GetLatestBlockHeader(ctx context.Context, _ *access.GetLatestBlockHeaderRequest) (*access.BlockHeaderResponse, error) {
-	height, err := s.index.Last()
-	if err != nil {
-		return nil, fmt.Errorf("could not retrieve last block height: %w", err)
-	}
-
-	req := access.GetBlockHeaderByHeightRequest{
-		Height: height,
-	}
-
-	return s.GetBlockHeaderByHeight(ctx, &req)
-}
-
-// GetBlockHeaderByID implements the GetBlockHeaderByID endpoint from the Flow Access API.
-// See https://docs.onflow.org/access-api/#getblockheaderbyid
-func (s *Server) GetBlockHeaderByID(ctx context.Context, in *access.GetBlockHeaderByIDRequest) (*access.BlockHeaderResponse, error) {
-	blockID := flow.HashToID(in.Id)
-	height, err := s.index.HeightForBlock(blockID)
-	if err != nil {
-		return nil, fmt.Errorf("could not get height for block %x: %w", blockID, err)
-	}
-
-	req := access.GetBlockHeaderByHeightRequest{
-		Height: height,
-	}
-
-	return s.GetBlockHeaderByHeight(ctx, &req)
-}
-
-// GetBlockHeaderByHeight implements the GetBlockHeaderByHeight endpoint from the Flow Access API.
-// See https://docs.onflow.org/access-api/#getblockheaderbyheight
-func (s *Server) GetBlockHeaderByHeight(_ context.Context, in *access.GetBlockHeaderByHeightRequest) (*access.BlockHeaderResponse, error) {
-	header, err := s.index.Header(in.Height)
-	if err != nil {
-		return nil, fmt.Errorf("could not retrieve block header at height %d: %w", in.Height, err)
-	}
-
-	decoder := signature.NewBlockSignerDecoder(nil)
```

Comment:
This is the root cause of the incomplete implementation. 

it requires the protocol state module in order to create the decoder to convert signer indices in the block header into signer IDs. However, it is much harder to do it in archive access, as it doesn't have access to the protocol state.